htdocs/core/class/html.form.class.php

```diff
@@ -3083,10 +3083,14 @@ public function select_produits_list($selected = 0, $htmlname = 'productid', $fi
 		} elseif ($finished == 1) {
 			$sql .= " AND p.finished = ".((int) $finished);
 		}
-		if ($status >= 0) {
+		if (getDolGlobalString('PRODUIT_FORCE_HIDE_NOT_FOR_SALE')) {
```

Comment:
There is already a param $status and $status_purchase so it is possible to decide if we want on purchase and/or on sale products depending on the context.

Adding option here will overwrite any context and break the feature to have it when we need, so it is better to set the value of status by the caller depending on context (only the caller can know if he want/need a status or not have them when he don't need/want).